lib/module.js

```diff
@@ -222,10 +225,6 @@ Module._findPath = function(request, paths, isMain) {
         } else {
           filename = toRealPath(basePath);
         }
-      } else if (rc === 1) {  // Directory.
-        if (exts === undefined)
-          exts = Object.keys(Module._extensions);
-        filename = tryPackage(basePath, exts, isMain);
       }
```

Comment:
👆 at the moment because the trailing slash check is for forward slash only, on Windows paths ending in a back slash hit that code path. Which means it will now fall through to the `tryExtensions` call on 234, instead of the `tryPackage` call on 243.